plaso/parsers/postgresql.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+"""Parser for PostgreSQL log files."""
+
+import string
+import pyparsing
+
+from dateutil import parser
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import definitions, errors
+from plaso.parsers import logger
+from plaso.parsers import manager
+from plaso.parsers import text_parser
+
+class PostgreSQLEventData(events.EventData):
+  """PostgreSQL log data.
+
+  Attributes:
+    pid (int): process identifier (PID).
+    user (str): "user@database" string if present.
+    log_level (str): logging level of event
+    log_line (str): log message.
+  """
+
+  DATA_TYPE = 'postgresql'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(PostgreSQLEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.pid = None
+    self.user = None
+    self.log_level = None
+    self.log_line = None
+
+class PostgreSQLParser(text_parser.PyparsingMultiLineTextParser):
+  """Parses events from PostgreSQL log files."""
+
+  NAME = 'postgresql'
+  DATA_FORMAT = 'PostgreSQL log file'
+  _ENCODING = 'utf-8'
+
+  _DATE_TIME = pyparsing.Group(
+    pyparsing.LineStart() +
```

Comment:
use 4 space hanging indents in these pyparsing definitions..

Reply:
Done